analyzers/tests/SonarAnalyzer.UnitTest/Extensions/SyntaxNodeExtensionsTest.cs

```diff
@@ -578,6 +578,47 @@ public void M()
             }
         }
 
+        [TestMethod]
+        public void IsInExpressionTree_CS()
+        {
+            const string code = @"
+using System.Linq;
+public class Sample
+{
+    public void Main(int[] arr)
+    {
+            var withNormalLambda = arr.Where(xNormal => xNormal == 42).OrderBy((xNormal) => xNormal).Select(xNormal => xNormal);
+            var withNormal = from xNormal in arr where xNormal == 42 orderby xNormal select xNormal;
+
+            var withExpressionLambda = arr.AsQueryable().Where(xExpres => xExpres == 42).OrderBy((xExpres) => xExpres).Select(xExpres => xExpres);
+            var withExpression = from xExpres in arr.AsQueryable() where xExpres == 42 orderby xExpres select xExpres;
+    }
+}";
+            var (tree, model) = TestHelper.CompileCS(code);
+            var allIdentifiers = tree.GetRoot().DescendantNodes().OfType<CS.IdentifierNameSyntax>().Where(x => x.Parent is not CS.SelectClauseSyntax).ToArray(); // SelectClause doesn't resolve symbol
```

Comment:
> // SelectClause doesn't resolve symbol

That is because of optimization during the lowering of the query syntax to "normal" syntax:
If select doesn't do anything special, the redundant `Select(x => x)` after the `OrderBy` is not "translated". If you do `xExpres.ToString()` it will be there.

Reply:
I did it everywhere (even when not needed for VB) to keep all the cases aligned and doing the same thing